### DISCUSSION TITLE
removing pnpm-lock.yaml causes nextjs standalone build output structure to differ

### DISCUSSION THREAD

**Discussion by vijayth2-cerebras** [vercel/next.js#71904] [Help]
### Summary

I have a Yarn workspace project with the following structure
```
workspaceRoot
  apps
    common
    web-common
    server-common
    chat
```
The chat project is a NextJS app (14.3). I inadvertently have a pnpm-lock.yaml in the chat project. With the pnpm-lock.yaml (even an empty one), the nextjs standalone build produces the following structure
```
.next
  standalone
    server.js
```
If I remove pnpm-lock.yaml, then the output structure is different
```
.next
  standalone
    common
    web-common
    server-common
    chat
      server.js
```
I would like to remove the pnpm-lock.yaml and still retain the simpler structure where server.js is directly within the standalone folder. Any help to resolve this weird behaviour is appreciated.

**Comment by aniruddhaadak80** [comment]
Hello, @vijayth2-cerebras 
---

It looks like the presence of `pnpm-lock.yaml` in your Next.js chat project is affecting the standalone build output. To keep the simpler structure with `server.js` directly in the `standalone` folder while removing the lock file, please try the following:

1. **Clear the .next Folder**: Run `rm -rf .next` to remove any cached build artifacts.
2. **Reinstall Dependencies**: Use Yarn to reinstall your dependencies by running `yarn install`.
3. **Build Again**: Finally, execute `yarn build` to see if the structure improves.

If the issue persists, it might be helpful to report it on the Next.js GitHub repository for more specific guidance.

I hope this helps!

---

**Reply by vijayth2-cerebras** [reply]
It happens in CI also. It is a fresh build in CI.

**Reply by aniruddhaadak80** [reply]
---

Hi @vijayth2-cerebras,

Thanks for the update! Since the issue occurs in a fresh CI build, here are a few things to check:

1. **CI Setup**: Make sure the CI environment is configured to use Yarn, and there are no leftover pnpm files affecting the build.

2. **Version Check**: Ensure the versions of Next.js and Yarn in CI match your local setup to avoid discrepancies.

3. **Lock File Generation**: Check if any scripts are creating the pnpm-lock.yaml during the CI build. You might need to remove it before the build starts.


Let me know if you need more help!

---
